### PR TITLE
Add Courses and StudentGroups to Mentor admin panel

### PR DESCRIPTION
**Pull request by Vlad1023**
Also, add endpoints to get mentor`s Courses and StudentGroups

### CODE REVIEW

**Review by SerjKocherga — APPROVED** [review]

File: CharlieBackend.AdminPanel/Services/MentorService.cs [hunk 1]
```diff
@@ -57,9 +58,14 @@ public async Task<IList<MentorViewModel>> GetAllMentorsAsync(string accessToken)
 
         public async Task<MentorEditViewModel> GetMentorByIdAsync(long id, string accessToken)
         {
-            var mentor = await
-                            _apiUtil.GetAsync<MentorEditViewModel>($"{_config.Value.Urls.Api.Https}/api/mentors/{id}", accessToken);
-    
+            var mentorTask = _apiUtil.GetAsync<MentorEditViewModel>($"{_config.Value.Urls.Api.Https}/api/mentors/{id}", accessToken);
+            var coursesTask = _apiUtil.GetAsync<IList<CourseViewModel>>($"{_config.Value.Urls.Api.Https}/api/courses", accessToken);
+            var studentGroupTask = _apiUtil.GetAsync<IList<StudentGroupViewModel>>($"{_config.Value.Urls.Api.Https}/api/student_groups", accessToken);
```
Comment (filinskyi) [inline CharlieBackend.AdminPanel/Services/MentorService.cs:63]: Why are temporary variables needed? Do inline instead.

**Review by filinskyi — CHANGES_REQUESTED** [review]

File: CharlieBackend.AdminPanel/Services/MentorService.cs (hunk 1 above)
Comment (BohdanBBG) [reply CharlieBackend.AdminPanel/Services/MentorService.cs:63]: Everything is correct, since this is an async API request

**Review by BohdanBBG — APPROVED** [review]

**Review by filinskyi — APPROVED** [review]

**Review by x1mx1x — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Business/Services/Interfaces/IMentorService.cs [hunk 2]
```diff
@@ -14,6 +14,10 @@ public interface IMentorService
 
         public Task<IList<MentorDto>> GetAllMentorsAsync();
 
+        public Task<Result<IList<MentorStudyGroupsDto>>> GetMentorStudyGroupsByMentorIdAsync(long id);
+
+        public Task<Result<IList<MentorCoursesDto>>> GetMentorCoursesByMentorIdAsync(long id);
+
         public Task<long?> GetAccountId(long mentorId);
 
         public Task<Result<MentorDto>> UpdateMentorAsync(long id, UpdateMentorDto mentorModel);
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/Interfaces/IMentorService.cs:23]: Keep interface members without default access modifiers

File: CharlieBackend.Data/Repositories/Impl/Interfaces/ICourseRepository.cs [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 
 namespace CharlieBackend.Data.Repositories.Impl.Interfaces
 {
     public interface ICourseRepository : IRepository<Course>
     {
         public Task<bool> IsCourseNameTakenAsync(string courseName);
 
+        public Task<List<MentorCoursesDto>> GetMentorCourses(long id);
+
         public Task<List<Course>> GetCoursesByIdsAsync(List<long> courseIds);
```
Comment (whereismyroot) [inline CharlieBackend.Data/Repositories/Impl/Interfaces/ICourseRepository.cs:14]: please keep interface members without access modifiers

File: CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs [hunk 4]
```diff
@@ -90,5 +91,17 @@ public new Task<StudentGroup> GetByIdAsync(long id)
                     .Include(group => group.MentorsOfStudentGroups)
                     .FirstOrDefaultAsync(group => group.Id == id);
         }
+
+        public async Task<List<MentorStudyGroupsDto>> GetMentorStudyGroups(long id)
+        {
+            return await _applicationContext.StudentGroups
+                    .Include(group => group.MentorsOfStudentGroups)
+                    .Where(x => x.MentorsOfStudentGroups.Any(x => x.MentorId == id))
```
Comment (whereismyroot) [inline CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs:99]: Any checks for active/inactive group, course?

File: CharlieBackend.Business/Services/MentorService.cs [hunk 5]
```diff
@@ -180,5 +181,39 @@ public async Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync()
...
+            {
+                return Result<IList<MentorCoursesDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundCourses = await _unitOfWork.CourseRepository.GetMentorCourses(id);
+
+            if (!foundCourses.Any())
+            {
+                return Result<IList<MentorCoursesDto>>.GetError(ErrorCode.NotFound, $"Courses for mentor with id {id} not found");
+            }
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/MentorService.cs:214]: just return empty set

File: CharlieBackend.Business/Services/MentorService.cs [hunk 6]
```diff
@@ -180,5 +181,39 @@ public async Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync()
...
+            {
+                return Result<IList<MentorStudyGroupsDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundGroups = await _unitOfWork.StudentGroupRepository.GetMentorStudyGroups(id);
+
+            if (!foundGroups.Any())
+            {
+                return Result<IList<MentorStudyGroupsDto>>.GetError(ErrorCode.NotFound, $"Study groups for mentor with id {id} not found");
+            }
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/MentorService.cs:197]: Just return empty set

**Review by whereismyroot — CHANGES_REQUESTED** [review]
small change

File: CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs (hunk 4 above)
Comment (Vlad1023) [reply CharlieBackend.Data/Repositories/Impl/StudentGroupRepository.cs:99]: Is it needed?

File: CharlieBackend.Business/Services/MentorService.cs (hunk 6 above)
Comment (Vlad1023) [reply CharlieBackend.Business/Services/MentorService.cs]: Done. Refcatored to return empty list

File: CharlieBackend.Business/Services/MentorService.cs (hunk 5 above)
Comment (Vlad1023) [reply CharlieBackend.Business/Services/MentorService.cs]: Done. Refcatored to return empty list

File: CharlieBackend.Data/Repositories/Impl/Interfaces/ICourseRepository.cs (hunk 3 above)
Comment (Vlad1023) [reply CharlieBackend.Data/Repositories/Impl/Interfaces/ICourseRepository.cs]: done

File: CharlieBackend.Business/Services/Interfaces/IMentorService.cs (hunk 2 above)
Comment (Vlad1023) [reply CharlieBackend.Business/Services/Interfaces/IMentorService.cs]: done

File: CharlieBackend.Business/Services/MentorService.cs [hunk 7]
```diff
@@ -180,5 +181,41 @@ public async Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync()
...
+                return Result<IList<MentorCoursesDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundCourses = await _unitOfWork.CourseRepository.GetMentorCourses(id);
+
+            if (!foundCourses.Any())
+            {
+                return Result<IList<MentorCoursesDto>>.
+                    GetSuccess(new List<MentorCoursesDto>());
+            }
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/MentorService.cs:216]: You don't need this if at all. repository returns empty list if nothing was found, and you can pass it to line 218

File: CharlieBackend.Business/Services/MentorService.cs [hunk 8]
```diff
@@ -180,5 +181,41 @@ public async Task<Result<IList<MentorDto>>> GetAllActiveMentorsAsync()
...
+                return Result<IList<MentorStudyGroupsDto>>.GetError(ErrorCode.NotFound, "Mentor doesn`t exist");
+            }
+
+            var foundGroups = await _unitOfWork.StudentGroupRepository.GetMentorStudyGroups(id);
+
+            if (!foundGroups.Any())
+            {
+                return Result<IList<MentorStudyGroupsDto>>.
+                    GetSuccess(new List<MentorStudyGroupsDto>());
+            }
```
Comment (whereismyroot) [inline CharlieBackend.Business/Services/MentorService.cs:198]: You don't need this if at all. repository returns empty list if nothing was found, and you can pass it to line 218

**Review by whereismyroot — CHANGES_REQUESTED** [review]
Slight change

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**